### PR TITLE
CDD-1192 api definition

### PR DESCRIPTION
**Pull request by stewabel**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/inventorylinking/imports/views/definition.scala.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "description": "Single WCO-compliant Customs Inventory Linking Import Declaration API",
+    "context": "customs/inventory-linking/imports",
+    "versions": [
+      {
+        "version": "1.0",
+        "status": "PUBLISHED",
+        "access": {
+          "type": "PUBLIC"
+        }
+      }
```
Comment (cichockimc) [inline app/uk/gov/hmrc/customs/inventorylinking/imports/views/definition.scala.txt]: no `fieldDefinitions` ?

File: conf/application.conf [hunk 2]
```diff
@@ -59,10 +59,9 @@ application.langs="en"
 # ~~~~~
 # Define the Global object class for this application.
 # Default to Global in the root package.
-application.global = CustomsApiGlobal
+#application.global = CustomsApiGlobal
```
Comment (cichockimc) [inline conf/application.conf]: shall we remove this line?

File: conf/application.conf [hunk 3]
```diff
@@ -59,10 +59,9 @@ application.langs="en"
...
 
 play.http.requestHandler = "play.api.http.GlobalSettingsHttpRequestHandler"
-play.http.errorHandler = "CustomErrorHandler"
```
Comment (cichockimc) [inline conf/application.conf]: will error handling work as expected without this line (in terms of responses) ?

File: conf/dynamicDefinition.routes [hunk 4]
```diff
@@ -0,0 +1,2 @@
+GET    /api/definition                              @uk.gov.hmrc.customs.inventorylinking.imports.controllers.DocumentationController.definition()
+#GET    /api/conf/:version/*file                     @uk.gov.hmrc.customs.inventorylinking.imports.controllers.DocumentationController.conf(version, file)
```
Comment (cichockimc) [inline conf/dynamicDefinition.routes]: I guess it'll be implemented later ?

File: public/api/conf/1.0/application.raml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        body:
+          application/xml:
+        responses:
+          204:
+          400:
+            body:
+              application/xml:
+                description: Invalid XML Payload
+                type: !include schemas/customs/error.xsd
+                example: |
```
Comment (cichockimc) [inline public/api/conf/1.0/application.raml]: will it work this way without `CustomErrorHandler` ?

File: conf/application.conf (hunk 3 above)
Comment (stewabel) [reply conf/application.conf]: I need to discuss migration strategy of customs-api-common into open.

File: conf/dynamicDefinition.routes (hunk 4 above)
Comment (stewabel) [reply conf/dynamicDefinition.routes]: Yes, again customs-api-common is a bit of a blocker at the moment

File: app/uk/gov/hmrc/customs/inventorylinking/imports/views/definition.scala.txt (hunk 1 above)
Comment (stewabel) [reply app/uk/gov/hmrc/customs/inventorylinking/imports/views/definition.scala.txt]: We have separate story for field definitions - CDD-1197

File: public/api/conf/1.0/application.raml (hunk 5 above)
Comment (stewabel) [reply public/api/conf/1.0/application.raml]: No, however, in a subsequent PR we will implement error handling, and decide at that point

**Comment by cichockimc** [comment]
:+1:

**Pull request merged**